library/Faktory/Job.hs

```diff
@@ -101,19 +104,23 @@ jobRetriesRemaining job = max 0 $ enqueuedRetry - attemptCount
   enqueuedRetry = maybe faktoryDefaultRetry getLast $ joRetry $ jobOptions job
   attemptCount = maybe 0 ((+ 1) . jfRetryCount) $ jobFailure job
 
+jobReserveFor :: Job arg -> Maybe Natural
+jobReserveFor = fmap getLast . joReserveFor . jobOptions
+
 instance ToJSON args => ToJSON (Job args) where
   toJSON = object . toPairs
   toEncoding = pairs . mconcat . toPairs
 
 toPairs :: (KeyValue a, ToJSON arg) => Job arg -> [a]
-toPairs Job {..} =
+toPairs job@Job {..} =
   [ "jid" .= jobJid
   , "at" .= jobAt
   , "args" .= jobArgs
   , "jobtype" .= joJobtype jobOptions
   , "retry" .= joRetry jobOptions
   , "queue" .= joQueue jobOptions
   , "custom" .= joCustom jobOptions
+  , "reserve_for" .= jobReserveFor job
```

Comment:
Functionally equivalent, but there's no need to unwrap the `Last` to encode this JSON. I *think* it might be better to transparently use the `JobOptions` accessor, just in-case the `Job`-level accessor grows more decoration that shouldn't get sucked into encoding. It at least feels more consistent with the rest of these here. Take it or leave it though.

```suggestion
  , "reserve_for" .= joReserveFor jobOptions
```